tutorials/03_moduly.md

```diff
@@ -0,0 +1,346 @@
+moduly
+======
+
+Zatím jsme tvořili programy v Pythonu tak nějak na divoko, tedy v jednom nebo
+více souborech bez nějakého zvláštního řádu. Na dnešním cvičení se podíváme na
+to, jak tvořit redistribuovatelné moduly, které jdou instalovat pomocí pipu.
+
+Za příklad si vezmeme kód Ondřeje Caletky, který umožňuje určit české svátky
+v zadaném roce. Jako příklad je ideální, protože obsahuje jak funkce, které
+můžeme volat z Pythonu, tak lze volat z příkazové řádky.
+
+ * [oskar456/isholiday.py](https://gist.github.com/oskar456/e91ef3ff77476b0dbc4ac19875d0555e)
+
+setup.py
+--------
+
+Základním stavebním kamenem Python balíčku je soubor `setup.py`, který
+obsahuje všechna potřebná metadata ve volání funkce `setup()` z modulu
+`setuptools`.
+
+Pojďme vytvořit jeho minimální variantu:
+
+```python
+from setuptools import setup
+
+
+setup(
+    name='isholiday',
+    version='0.1',
+    description='Finds Czech holiday for given year',
+    author='Ondřej Caletka',
+    author_email='ondrej@caletka.cz',
+    license='Public Domain',
+    url='https://gist.github.com/oskar456/e91ef3ff77476b0dbc4ac19875d0555e',
+    py_modules=['isholiday'],
+)
+```
+
+Všimněte si, že jsme balíček pojmenoval stejně jako soubor se zdrojovým kódem.
```

Comment:
Tahle informace mi přijde neúplná, pro začátečníky možná zavádějící. Podstatné je, že se balíček jmenuje stejně jako základní _modul_. Tohle je pouze speciální případ, kdy balíček obsahuje pouze jeden soubor, a tedy to odpovídá názvu toho souboru, nikoli adresáře modulu, což je běžnější.